device/ospf/ospfv2/interface/auth.yaml

```diff
@@ -0,0 +1,42 @@
+components:
+  schemas:
+    Ospfv2Interface.Authentication:
+      description: |-
+        This contains OSPFv4 authentication properties.
+        Reference: https://www.rfc-editor.org/rfc/rfc2328#appendix-D
+      type: object
+      required: [auth_type]
+      properties:
+        auth_type:
+          description: |-
+            The authentication method.
+            - md5: If the authentication type is of 'md5' then 'md5_key_id' and 'md5_key' both are to be configured.
+            - password: 
+          type: string
+          x-field-uid: 1
+          x-enum:
+            md5:
+              x-field-uid: 1
+            password:
+              x-field-uid: 2
+        md5_key_id:
+          description: |-
+            The unique MD5 Key Identifier per-interface.
+          type: string
```

Comment:
I think this is an integer not a string ... and we should add text from closer to RFC .. intention is to indicate to the receiver which of a set of MD5 keys have been used in calculating the MD5 hash added to the received OSPFv2 PDU.